packages/native/src/useLinking.tsx

```diff
@@ -592,7 +592,13 @@ export default function useLinking(
               // We couldn't find an existing entry to go back to, so we'll go back by the delta
               // This won't be correct if multiple routes were pushed in one go before
               // Usually this shouldn't happen and this is a fallback for that
-              await history.go(historyDelta);
+
+              // If window.history.length is big enough then it's ok to go back. Otherwise,
+              // we can't do window.history.go(historyDelta) because that will cause the
+              // browser to leave the page/application.
+              if (window.history.length + historyDelta >= 2) {
```

Comment:
`window.history.length` isn't a reliable way to check if we can go back in history. it contains the history of all the navigation in the tab - including pages from different URLs, if there are any pages in the forward stack, etc., and doesn't get reset after refreshing or going back.

the only way to check it is with the `items` array that we have created.

Reply:
Ok yeah sorry, that's totally true.

So if I save the browser history length upon page load, then I can use that value in the condition:
```
if (window.history.length + historyDelta >= initialBrowserHistoryLength) {
  await history.go(historyDelta);
}
```

Might that be the solution?

Reply:
> So if I store the browser history length upon page load

`history.length` also includes the forward stack like I mentioned, not just previous items - which means it won't return what you expect if the user presses the browser's back button at any point. `history.length` shouldn't be used at all.

A potential workaround to the relevant issue is to push new entries to the history if React Naigation's history has more than one item on page load. It can be done around like 527 where the URL is updated on initial load with `history.replace`.

Reply:
Ok makes sense. So can we just do nothing if we arrive at this point in the code? In other words:
```
...

if (nextIndex !== -1 && nextIndex < currentIndex) {
  // An existing entry for this path exists and it's less than current index, go back to that
  await history.go(nextIndex - currentIndex);
} else {
  // We couldn't find an existing entry to go back to...

  // So do nothing because, given that historyDelta is negative, we're in danger of leaving the page
}

...
```

Or are you looking for a solution that addresses the root cause of why it ends up in a situation where there's no existing entry to go back to?

Reply:
I already suggested a workaround, please use that. It's not possible to address why it ends up in this branch - you already have a repro for when it happens and it can happen in many scenarios where history isn't linear.

Reply:
Ok thanks @satya164.